docroot/modules/humsci/hs_blocks/src/Plugin/Block/SocialMediaBlock.php

```diff
@@ -0,0 +1,305 @@
+    ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $plugin_id,
+    $plugin_definition
+    ) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration(): array {
+    return [
+      'icon_size' => 'small',
+      'layout' => 'grid',
+      'links' => [],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state): array {
+    $form['icon_size'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Icon Size'),
+      '#options' => [
+        'small' => $this->t('Small (32px)'),
+        'large' => $this->t('Large (48px)'),
+      ],
+      '#default_value' => $this->configuration['icon_size'],
+      '#required' => TRUE,
+    ];
+
+    $form['layout'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Layout'),
+      '#options' => [
+        'grid' => $this->t('Grid (no visible label)'),
+        'vertical_list' => $this->t('Vertical List (with visible label)'),
+      ],
+      '#default_value' => $this->configuration['layout'],
+      '#required' => TRUE,
+    ];
+
+    $form['links'] = [
+      '#type' => 'multivalue',
+      '#title' => $this->t('Links'),
+      '#description' => $this->t('Popular social platforms will show their icon, otherwise a generic icon will be shown.'),
+      '#cardinality' => MultiValue::CARDINALITY_UNLIMITED,
+      '#default_value' => ($this->configuration['links']) ? $this->configuration['links'] : [],
+      '#element_validate' => [
+        [get_class($this), 'validateLinks'],
+      ],
+      'link_url' => [
+        '#type' => 'url',
+        '#title' => $this->t('URL'),
+        '#description' => $this->t('Social Media Profile URL.'),
+      ],
+      'link_title' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Label'),
+        '#description' => $this->t('If empty, the social platform name will be used for popular platforms. If the platform is unknown then the domain name will be used.'),
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state): void {
+    $this->configuration['icon_size'] = $form_state->getValue('icon_size');
+    $this->configuration['layout'] = $form_state->getValue('layout');
+
+    // Only save links if they have data.
+    $links = $form_state->getValue('links');
+    $filtered_links = [];
+    foreach ($links as $link) {
+      if (!empty($link['link_url'] || !empty($link['link_title']))) {
+        $filtered_links[] = $link;
+      }
+    }
+    $this->configuration['links'] = $filtered_links;
```

Comment:
```suggestion
    // Only save links if they have data.
    $links = [];
    foreach ($form_state->getValue('links') as $link) {
      if (!empty($link['link_url'] || !empty($link['link_title']))) {
        $links[] = $link;
      }
    }
    $this->configuration['links'] = $links;
```

Also quick question on this -- should a user be able to save a social media icon link without a URL just because it has a title? It also seems like the validation method would prevent such a scenario in the first place.

Reply:
Yes, the validation prevents from saving items with only title. I changed it to check only for the URL.